lib/ravelin/ravelin_object.rb

```diff
@@ -0,0 +1,50 @@
+module Ravelin
+  class RavelinObject
+    class << self
+      attr_reader :attributes
+
+      def required_attributes
+        @required_attributes ||= []
+      end
+
+      def attr_required(*names)
+        @required_attributes = *names
+      end
+
+      def attr_accessor(*names)
```

Comment:
The Ruby `attr_accessor` doesn't let you access the arguments passed into it. I overwrote it to add the additional functionality of assigning the arguments to the `@attributes` variable.

The `serializable_hash` method can then use the `@attributes` to build a hash using this list of arguments.